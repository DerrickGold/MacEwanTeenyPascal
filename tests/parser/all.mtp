(* constants *)
const test :=10;
      floop := 1337;
      three := 'Stringify';

(* variable declaration *)
var newVar : integer; otherVar : array(20) of integer;
    sample1, sample2, andThree : integer;


begin
(* statement lists in here*)

	(*
	 * nested block statement
	 * These variables are only accessible in the nested block
	 *)
	var sample1, sample2, andThree : array(test) of integer;
	begin

	end;
	(* this should also produce statements *)
	sample1 := not 10 * 20;


	case sample2 of
		20 :
		   andThree := 30
		;
		40 :
		   andThree := 25
	end;

	
	case andThree of
		sample2 : andThree := 45
	else
		sample2 := 0
	end;

	
	if (andThree = 25) then
	begin
		sample1 := 0
	end;

	if (andThree = 11) then
		sample2 := 100
	else
		sample1 := 33;

	while (andThree) do
		sample1 := sample1 + 2;


	write(andThree);
	write(andThree, sample2, sample1, newVar);
	write(andThree, sample2);

	read(andThree);
	read(andThree, sample2, sample1, newVar);
	read(andThree, sample2)
end
.