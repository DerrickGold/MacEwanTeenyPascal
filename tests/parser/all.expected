[Head] (Block)
    [Child] (Const Section)
        [Child] (Const Declaration)
            [Child] 2, TOK_ID, test, (Literal)
            [Child] 2, TOK_NUM, 10, (Literal)
        [Sibling] (Const Declaration)
            [Child] 3, TOK_ID, floop, (Literal)
            [Child] 3, TOK_NUM, 1337, (Literal)
        [Sibling] (Const Declaration)
            [Child] 4, TOK_ID, three, (Literal)
            [Child] 4, TOK_STR, 'Stringify', (Literal)
    [Child] (Variable Section)
        [Child] (Variable Declaration, Variable List)
            [Child] 7, TOK_ID, newVar, (Literal)
            [Child] 7, TOK_KEY_INTEGER, integer, (Type)
        [Sibling] (Variable Declaration, Variable List)
            [Child] 7, TOK_ID, otherVar, (Literal)
            [Child] 7, TOK_NUM, 20, (Type, Constant, Array, Integer)
    [Child] (Statement List)
        [Child] (Block, Block Statement)
            [Child] (Variable Section)
                [Child] (Variable Declaration, Variable List)
                    [Child] 14, TOK_ID, sample1, (Literal)
                    [Sibling] 14, TOK_ID, sample2, (Literal)
                    [Sibling] 14, TOK_ID, andThree, (Literal)
                    [Child] 14, TOK_ID, newVar, (Type, Constant, Simple Name, Array, Integer)
            [Child] (Statement List)
                [Child] (Null Statement)
        [Sibling] (Assign Statement)
            [Child] 24, TOK_ID, sample1, (Simple Name)
            [Child] 24, TOK_STAR, *, (Expression, Simple Expression, Multiply Operator)
                [Child] 24, TOK_NUM, 10, (Factor, Constant, Term, Not)
                [Child] 24, TOK_NUM, 20, (Factor, Constant, Term)
        [Sibling] (Case Statement)
            [Child] 27, TOK_ID, sample2, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Child] (Case List)
                [Child] (Case)
                    [Child] (Const List)
                        [Child] 28, TOK_NUM, 20, (Constant)
                    [Child] (Assign Statement)
                        [Child] 28, TOK_ID, andThree, (Simple Name)
                        [Child] 28, TOK_NUM, 30, (Expression, Simple Expression, Factor, Constant)
                [Sibling] (Case)
                    [Child] (Const List)
                        [Child] 29, TOK_NUM, 40, (Constant)
                    [Child] (Assign Statement)
                        [Child] 29, TOK_ID, andThree, (Simple Name)
                        [Child] 29, TOK_NUM, 25, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (Case Statement)
            [Child] 33, TOK_ID, andThree, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Child] (Case List)
                [Child] (Case)
                    [Child] (Const List)
                        [Child] 34, TOK_ID, sample2, (Constant, Simple Name)
                    [Child] (Assign Statement)
                        [Child] 34, TOK_ID, andThree, (Simple Name)
                        [Child] 34, TOK_NUM, 45, (Expression, Simple Expression, Factor, Constant)
            [Child] (Assign Statement)
                [Child] 36, TOK_ID, sample2, (Simple Name)
                [Child] 36, TOK_NUM, 0, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 40, TOK_EQ, =, (Condition, Expression, Simple Expression, Factor, Relational Operator)
                [Child] 40, TOK_ID, andThree, (Simple Expression, Factor, Constant, Simple Name)
                [Child] 40, TOK_NUM, 25, (Simple Expression, Factor, Constant)
            [Child] (Block, Block Statement)
                [Child] (Statement List)
                    [Child] (Assign Statement)
                        [Child] 42, TOK_ID, sample1, (Simple Name)
                        [Child] 42, TOK_NUM, 0, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (If Statement)
            [Child] 45, TOK_EQ, =, (Condition, Expression, Simple Expression, Factor, Relational Operator)
                [Child] 45, TOK_ID, andThree, (Simple Expression, Factor, Constant, Simple Name)
                [Child] 45, TOK_NUM, 11, (Simple Expression, Factor, Constant)
            [Child] (Assign Statement)
                [Child] 46, TOK_ID, sample2, (Simple Name)
                [Child] 46, TOK_NUM, 100, (Expression, Simple Expression, Factor, Constant)
            [Child] (Assign Statement)
                [Child] 48, TOK_ID, sample1, (Simple Name)
                [Child] 48, TOK_NUM, 33, (Expression, Simple Expression, Factor, Constant)
        [Sibling] (While Statement)
            [Child] 50, TOK_ID, andThree, (Condition, Expression, Simple Expression, Factor, Constant, Simple Name)
            [Child] (Assign Statement)
                [Child] 51, TOK_ID, sample1, (Simple Name)
                [Child] 51, TOK_PLUS, +, (Expression, Simple Expression, Binary Adding Operator)
                    [Child] 51, TOK_ID, sample1, (Simple Expression, Factor, Constant, Simple Name)
                    [Child] 51, TOK_NUM, 2, (Factor, Constant)
        [Sibling] (Write Statement)
            [Child] 54, TOK_ID, andThree, (Expression, Simple Expression, Factor, Constant, Simple Name)
        [Sibling] (Write Statement)
            [Child] 55, TOK_ID, andThree, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Sibling] 55, TOK_ID, sample2, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Sibling] 55, TOK_ID, sample1, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Sibling] 55, TOK_ID, newVar, (Expression, Simple Expression, Factor, Constant, Simple Name)
        [Sibling] (Write Statement)
            [Child] 56, TOK_ID, andThree, (Expression, Simple Expression, Factor, Constant, Simple Name)
            [Sibling] 56, TOK_ID, sample2, (Expression, Simple Expression, Factor, Constant, Simple Name)
        [Sibling] (Read Statement)
            [Child] 58, TOK_ID, andThree, (Simple Name)
        [Sibling] (Read Statement)
            [Child] 59, TOK_ID, andThree, (Simple Name)
            [Sibling] 59, TOK_ID, sample2, (Simple Name)
            [Sibling] 59, TOK_ID, sample1, (Simple Name)
            [Sibling] 59, TOK_ID, newVar, (Simple Name)
        [Sibling] (Read Statement)
            [Child] 60, TOK_ID, andThree, (Simple Name)
            [Sibling] 60, TOK_ID, sample2, (Simple Name)
